pinot-core/src/main/java/com/linkedin/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java

```diff
@@ -33,13 +35,40 @@
   private static Logger LOGGER = LoggerFactory.getLogger(ServerSegmentCompletionProtocolHandler.class);
   private static final int SEGMENT_UPLOAD_REQUEST_TIMEOUT_MS = 30_000;
   private static final int OTHER_REQUESTS_TIMEOUT = 5_000;
+  private static final String HTTPS_PROTOCOL = "https";
+  private static final String HTTP_PROTOCOL = "http";
+
+  private static final String CONFIG_OF_CONTROLLER_HTTPS_ENABLED = "https.enabled";
```

Comment:
I would recommend making these config keys into Constants class, let both fetcher and uploader share them (using different prefix).

Reply:
i will leave it as it is. the config names can be confusing

Reply:
let us discuss in person to finalize this